lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -2786,3 +2786,94 @@ def test_group_access(self, role_name, course_is_cohorted, thread_group_state):
             self.assertFalse(expected_error)
         except Http404:
             self.assertTrue(expected_error)
+
+@ddt.ddt
+class RetrieveThreadTest(
+        CommentsServiceMockMixin,
+        UrlResetMixin,
+        SharedModuleStoreTestCase
+):
+    """Tests for get_thread"""
+    @classmethod
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUpClass(cls):
+        super(RetrieveThreadTest, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {"ENABLE_DISCUSSION_SERVICE": True})
+    def setUp(self):
+        super(RetrieveThreadTest, self).setUp()
+        httpretty.reset()
+        httpretty.enable()
+        self.addCleanup(httpretty.disable)
+        self.user = UserFactory.create()
+        self.register_get_user_response(self.user)
+        self.request = RequestFactory().get("/test_path")
+        self.request.user = self.user
```

Comment:
I think it would be clearer if you did not assign `self.request.user` here, since some tests (`test_not_enrolled` for example) assign a different user. the setUp method should only contain things that are used across all the tests in this class.

And to remove some ambiguity, perhaps rename `self.user` to `self.thread_author`?